Cython/Utility/AsyncGen.c

```diff
@@ -416,8 +416,8 @@ static PyType_Spec __pyx_AsyncGenType_spec = {
     __PYX_TYPE_MODULE_PREFIX "async_generator",
     sizeof(__pyx_PyAsyncGenObject),
     0,
-#if PY_VERSION_HEX >= 0x030A0000
-    Py_TPFLAGS_IMMUTABLETYPE |
+#if __PYX_LIMITED_VERSION_HEX >= 0x030A0000
+    Py_TPFLAGS_IMMUTABLETYPE | Py_TPFLAGS_DISALLOW_INSTANTIATION |
```

Comment:
Hmm, but this means that we won't set the flag in the abi3 wheels. I think we need to define both flags ourselves if they are not defined yet.